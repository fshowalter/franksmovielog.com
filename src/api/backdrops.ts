import { getImage } from "astro:assets";
import path from "node:path";
import sharp from "sharp";

import { normalizeSources } from "./utils/normalizeSources";

export type BackdropImageProps = {
  src: string;
  srcSet: string;
};

const images = import.meta.glob<{ default: ImageMetadata }>(
  "/content/assets/backdrops/*.png",
);

export async function getBackdropImageProps(
  slug: string,
  {
    height,
    width,
  }: {
    height: number;
    width: number;
  },
): Promise<BackdropImageProps> {
  const backdropFile = await getBackdropFile(slug);

  const optimizedImage = await getImage({
    format: "avif",
    height: height,
    quality: 80,
    src: backdropFile.default,
    width: width,
    widths: [0.3, 0.5, 0.8, 1].map((w) => w * width),
  });

  return {
    src: normalizeSources(optimizedImage.src),
    srcSet: normalizeSources(optimizedImage.srcSet.attribute),
  };
}

export async function getOpenGraphBackdropAsBase64String(slug: string) {
  const imageBuffer = await sharp(
    path.resolve(`./content/assets/backdrops/${slug}.png`),
  )
    .resize(1200)
    .toFormat("png")
    .toBuffer();

  return `data:${"image/png"};base64,${imageBuffer.toString("base64")}`;
}

async function getBackdropFile(slug: string) {
  let backdropFilePath = Object.keys(images).find((path) => {
    return path.endsWith(`${slug}.png`);
  })!;

  backdropFilePath =
    backdropFilePath ||
    Object.keys(images).find((path) => {
      return path.endsWith(`default.png`);
    })!;

  return await images[backdropFilePath]();
}
