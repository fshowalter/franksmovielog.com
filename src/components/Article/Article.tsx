import type { BackdropImageProps } from "src/api/backdrops";
import type { MoreReviewsValue } from "src/components/MoreReviews";

import { Backdrop } from "src/components/Backdrop";
import { Layout } from "src/components/Layout";
import { LongFormText } from "src/components/LongFormText";
import { MoreReviews } from "src/components/MoreReviews";
import { SubHeading } from "src/components/SubHeading";

export interface Props {
  backdropImageProps: BackdropImageProps;
  content: null | string;
  deck: string;
  recentReviews: MoreReviewsValue[];
  title: string;
}

export function Article({
  backdropImageProps,
  content,
  deck,
  recentReviews,
  title,
}: Props): JSX.Element {
  return (
    <Layout>
      <article>
        <Backdrop
          deck={deck}
          imageProps={backdropImageProps}
          size="large"
          title={title}
        />
        <section className="flex flex-col items-center pb-32 pt-16">
          <div className="px-container">
            <LongFormText className="max-w-prose" text={content} />
          </div>
        </section>
      </article>
      <div className="bg-subtle pb-16 pt-6">
        <MoreReviews values={recentReviews}>
          <SubHeading as="h2">
            Recent{" "}
            <a className="text-accent" href="/reviews/">
              Reviews
            </a>
          </SubHeading>
        </MoreReviews>
      </div>
    </Layout>
  );
}
