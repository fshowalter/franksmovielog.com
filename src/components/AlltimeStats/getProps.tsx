import { alltimeStats } from "src/api/alltimeStats";
import { getBackdropImageProps } from "src/api/backdrops";
import { getFluidWidthPosterImageProps } from "src/api/posters";
import { allStatYears } from "src/api/yearStats";
import { BackdropImageConfig } from "src/components/Backdrop";
import { MostWatchedMoviesPosterConfig } from "src/components/MostWatchedMovies";

import { type Props } from "./AlltimeStats";

export async function getProps(): Promise<Props> {
  const stats = await alltimeStats();
  const distinctStatYears = await allStatYears();

  return {
    backdropImageProps: await getBackdropImageProps(
      "stats",
      BackdropImageConfig,
    ),
    deck: `${(distinctStatYears.length - 1).toString()} Years in Review`,
    distinctStatYears,
    mostWatchedDirectors: await Promise.all(
      stats.mostWatchedDirectors.map(async (person) => {
        return {
          ...person,
          viewings: await Promise.all(
            person.viewings.map(async (viewing) => {
              return {
                ...viewing,
                posterImageProps: await getFluidWidthPosterImageProps(
                  viewing.slug,
                  MostWatchedMoviesPosterConfig,
                ),
              };
            }),
          ),
        };
      }),
    ),
    mostWatchedMovies: await Promise.all(
      stats.mostWatchedTitles.map(async (title) => {
        return {
          ...title,
          posterImageProps: await getFluidWidthPosterImageProps(
            title.slug,
            MostWatchedMoviesPosterConfig,
          ),
        };
      }),
    ),
    mostWatchedPerformers: await Promise.all(
      stats.mostWatchedPerformers.map(async (person) => {
        return {
          ...person,
          viewings: await Promise.all(
            person.viewings.map(async (viewing) => {
              return {
                ...viewing,
                posterImageProps: await getFluidWidthPosterImageProps(
                  viewing.slug,
                  MostWatchedMoviesPosterConfig,
                ),
              };
            }),
          ),
        };
      }),
    ),
    mostWatchedWriters: await Promise.all(
      stats.mostWatchedWriters.map(async (person) => {
        return {
          ...person,
          viewings: await Promise.all(
            person.viewings.map(async (viewing) => {
              return {
                ...viewing,
                posterImageProps: await getFluidWidthPosterImageProps(
                  viewing.slug,
                  MostWatchedMoviesPosterConfig,
                ),
              };
            }),
          ),
        };
      }),
    ),
    stats,
  };
}
