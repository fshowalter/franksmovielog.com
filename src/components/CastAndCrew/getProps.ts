import { getAvatarImageProps } from "~/api/avatars";
import { getBackdropImageProps } from "~/api/backdrops";
import { allCastAndCrew } from "~/api/castAndCrew";
import { BackdropImageConfig } from "~/components/Backdrop";
import { ListItemAvatarImageConfig } from "~/components/ListItemAvatar";

import type { Props } from "./CastAndCrew";
import type { ListItemValue } from "./CastAndCrew";

export async function getProps(): Promise<Props & { metaDescription: string }> {
  const { castAndCrew } = await allCastAndCrew();

  const values = await Promise.all(
    castAndCrew.map(async (member) => {
      const value: ListItemValue = {
        avatarImageProps: await getAvatarImageProps(
          member.slug,
          ListItemAvatarImageConfig,
        ),
        creditedAs: member.creditedAs,
        name: member.name,
        reviewCount: member.reviewCount,
        slug: member.slug,
      };

      return value;
    }),
  );

  return {
    backdropImageProps: await getBackdropImageProps(
      "cast-and-crew",
      BackdropImageConfig,
    ),
    deck: '"Round up the usual suspects."',
    initialSort: "name-asc",
    metaDescription:
      "A sortable index of directors, performers, and writers who contributed to the movies reviewed on this site. Sort by name or review count.",
    values,
  };
}
