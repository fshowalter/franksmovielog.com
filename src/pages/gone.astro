---
import { getOpenGraphBackdropSrc } from "src/api/backdrops";
import { Article } from "src/components/Article/Article";
import { getProps } from "src/components/Article/getProps";
import Layout from "src/layouts/Layout.astro";

const slug = "gone";

const props = await getProps({
  slug,
  alt: "Forget it, Jake. It's Chinatown.",
});

const openGraphImage = await getOpenGraphBackdropSrc(slug);
---

<Layout
  title="410: Gone"
  description="Forget it, Jake. It's Chinatown."
  image={openGraphImage}
  article
>
  <Article {...props} />
</Layout>
