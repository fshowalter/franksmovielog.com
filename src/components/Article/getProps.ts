import { getBackdropImageProps } from "~/api/backdrops";
import { getPage } from "~/api/pages";
import { loadExcerptHtml, mostRecentReviews } from "~/api/reviews";
import { getStillImageProps } from "~/api/stills";
import { BackdropImageConfig } from "~/components/Backdrop";
import { MoreReviewsImageConfig } from "~/components/MoreReviews";

import type { Props } from "./Article";

export async function getProps({
  deck,
  slug,
}: {
  deck: string;
  slug: string;
}): Promise<Props> {
  const { content, title } = await getPage(slug);
  const recentReviews = await mostRecentReviews(4);

  return {
    backdropImageProps: await getBackdropImageProps(slug, BackdropImageConfig),
    content,
    deck,
    recentReviews: await Promise.all(
      recentReviews.map(async (review) => {
        const titleWithExcerpt = await loadExcerptHtml(review);
        return {
          ...titleWithExcerpt,
          stillImageProps: await getStillImageProps(
            titleWithExcerpt.slug,
            MoreReviewsImageConfig,
          ),
        };
      }),
    ),
    title,
  };
}
