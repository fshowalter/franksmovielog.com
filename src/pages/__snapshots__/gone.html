<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width" />
    <link rel="icon" href="/favicon.ico" sizes="48x48" />
    <link rel="icon" href="/favicon.svg" sizes="any" type="image/svg+xml" />
    <link rel="apple-touch-icon" href="/apple-touch-icon.png" />
    <link rel="manifest" href="/site.webmanifest" />
    <link rel="sitemap" href="/sitemap-index.xml" />
    <meta name="description" content="Forget it, Jake. It's Chinatown." />
    <meta
      name="og:image"
      content="/_image?/content%2Fassets%2Fbackdrops%2Fgone.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&#38;w=1200&#38;h=675&#38;q=80&#38;f=jpeg"
    />
    <meta property="og:url" content="https://www.franksmovielog.com//" />
    <meta property="og:type" content="article" />
    <meta property="og:title" content="410: Gone" />
    <meta
      property="og:description"
      content="Forget it, Jake. It's Chinatown."
    />
    <title>410: Gone | Frank&#39;s Movie Log</title>
  </head>
  <body>
    <div>
      <a
        href="#content"
        class="absolute left-1/2 top-0.5 z-50 mx-auto bg-subtle px-6 py-2 text-center text-accent [transform:translate(-50%,calc(-100%_-_2px))] focus:[transform:translate(-50%,0%)]"
        >Skip to content</a
      >
      <div
        class="image-filter mx-auto min-h-4 w-full max-w-canvas bg-[url(&#x27;/assets/ripnotcomingsoon.avif&#x27;)] desktop:sticky desktop:top-0 desktop:z-40"
      ></div>
      <div class="mx-auto flex min-h-full max-w-canvas flex-col bg-default">
        <header
          class="flex flex-col items-center gap-6 bg-default px-pageMargin py-6 text-center shadow-bottom desktop:sticky desktop:top-4 desktop:z-40 desktop:flex-row desktop:flex-wrap desktop:justify-between desktop:py-8 desktop:text-left max:justify-center"
        >
          <div
            class="items-inherit justify-items-inherit flex flex-col max:absolute max:left-[var(--page-margin-width)]"
          >
            <h1
              class="whitespace-nowrap font-normal leading-8"
              style="font-size: 1.5625rem"
            >
              <a class="text-default" href="/">Frank&#x27;s Movie Log</a>
            </h1>
            <p class="w-full text-sm italic leading-4 text-muted desktop:pl-px">
              My life at the movies.
            </p>
          </div>
          <div
            class="w-full max-w-prose desktop:order-4 desktop:w-auto max:absolute max:right-[var(--page-margin-width)]"
          >
            <button
              data-open-modal="true"
              disabled=""
              aria-label="Search"
              aria-keyshortcuts="Control+K"
              class="safari-border-radius-fix desktop::w-auto desktop::ring-0 flex w-full items-center overflow-hidden rounded-md text-sm leading-6 text-subtle ring-1 ring-border hover:ring-accent"
            >
              <kbd
                class="ml-auto mt-px hidden min-h-6 flex-none items-center bg-subtle pl-3 pr-4 font-mono text-sm font-light opacity-70 desktop:flex"
                ><kbd class="text-md leading-5">Ctrl</kbd
                ><kbd class="text-sm">K</kbd></kbd
              ><svg
                class="mr-2 size-10 flex-none border-0 border-r border-default bg-subtle fill-subtle px-3 py-1 desktop:w-4 desktop:border-r-0 desktop:bg-unset desktop:px-0"
                viewBox="0 0 16 16"
              >
                <path
                  fill-rule="evenodd"
                  d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
                ></path>
                <path
                  fill-rule="evenodd"
                  d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
                ></path></svg
              ><span
                class="flex items-baseline whitespace-nowrap p-2 text-base opacity-70 desktop:hidden"
                >Quick search...</span
              >
            </button>
          </div>
          <nav class="w-full desktop:w-auto">
            <ul
              class="flex flex-wrap justify-center gap-x-4 gap-y-2 text-accent tablet:gap-x-6 desktop:justify-start"
            >
              <li class="block whitespace-nowrap tracking-0.5px">
                <a class="text-muted" href="/">Home</a>
              </li>
              <li class="block whitespace-nowrap tracking-0.5px">
                <a class="text-inherit" href="/how-i-grade/">How I Grade</a>
              </li>
              <li class="block whitespace-nowrap tracking-0.5px">
                <a class="text-inherit" href="/reviews/">Reviews</a>
              </li>
              <li class="block whitespace-nowrap tracking-0.5px">
                <a class="text-inherit" href="/viewings/">Viewing Log</a>
              </li>
              <li class="block whitespace-nowrap tracking-0.5px">
                <a class="text-inherit" href="/cast-and-crew/"
                  >Cast &amp; Crew</a
                >
              </li>
              <li class="block whitespace-nowrap tracking-0.5px">
                <a class="text-inherit" href="/collections/">Collections</a>
              </li>
              <li class="block whitespace-nowrap tracking-0.5px">
                <a class="text-inherit" href="/watchlist/">Watchlist</a>
              </li>
            </ul>
          </nav>
        </header>
        <div class="grow" id="content">
          <main>
            <article class="flex flex-col items-center">
              <h1
                data-pagefind-meta="title"
                class="text-[2rem] font-normal leading-none desktop:text-[2.25rem] px-pageMargin py-6 text-center desktop:py-8"
              >
                410 Gone
              </h1>
              <img
                srcset="
                  /_image?/content%2Fassets%2Fbackdrops%2Fgone.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=240&amp;q=80&amp;f=avif   240w,
                  /_image?/content%2Fassets%2Fbackdrops%2Fgone.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=480&amp;q=80&amp;f=avif   480w,
                  /_image?/content%2Fassets%2Fbackdrops%2Fgone.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=960&amp;q=80&amp;f=avif   960w,
                  /_image?/content%2Fassets%2Fbackdrops%2Fgone.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=1920&amp;q=80&amp;f=avif 1920w
                "
                src="/_image?/content%2Fassets%2Fbackdrops%2Fgone.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=960&amp;h=540&amp;q=80&amp;f=avif"
                width="960"
                height="540"
                sizes="(min-width: 960px) 960px, 100vw"
                loading="lazy"
                decoding="async"
                alt="Forget it, Jake. It&#x27;s Chinatown."
                class="mb-[5.33px]"
              />
              <div class="spacer-y-16"></div>
              <div class="px-pageMargin">
                <div
                  class="rendered-markdown font-light text-md/7 tracking-0.3px tablet:text-xl/8 max-w-prose"
                >
                  <p>Forget it, Jake. Itâ€™s Chinatown.</p>
                </div>
              </div>
              <div class="spacer-y-32"></div>
            </article>
            <div
              data-pagefind-ignore="true"
              class="flex w-full max-w-popout items-center justify-center bg-default tablet:max-w-full tablet:bg-subtle tablet:pb-32 tablet:pt-8"
            >
              <nav
                class="relative flex w-full flex-col items-center tablet:w-auto"
              >
                <div
                  class="w-full px-gutter py-2 shadow-bottom tablet:py-4 tablet:shadow-none desktop:px-pageMargin desktop:py-4"
                >
                  <span class="font-semibold text-muted">Latest<!-- --> </span
                  ><a class="text-accent" href="/reviews/">Reviews</a>
                </div>
                <ul
                  class="w-full tablet:grid tablet:w-auto tablet:grid-cols-[repeat(2,minmax(100px,312px))] tablet:gap-8 tablet:px-gutter desktop:max-w-unset desktop:grid-cols-[repeat(4,1fr)] desktop:px-pageMargin desktop:pt-2"
                >
                  <li
                    class="flow-root w-full px-gutter py-6 even:bg-subtle tablet:grid tablet:max-w-[312px] tablet:grid-rows-[auto_auto_1fr_auto] tablet:gap-y-2 tablet:overflow-hidden tablet:rounded-lg tablet:bg-default tablet:p-0 tablet:pb-8 tablet:shadow-all tablet:shadow-border tablet:even:bg-default"
                  >
                    <div class="row-start-1 row-end-1">
                      <a
                        href="/reviews/chamber-of-horrors-1966/"
                        class="safari-border-radius-fix float-right mb-2 ml-6 block w-[calc(50%_-_12px)] max-w-[312px] overflow-hidden rounded-lg tablet:float-none tablet:m-0 tablet:w-auto tablet:rounded-none"
                        ><img
                          srcset="
                            /_image?/content%2Fassets%2Fstills%2Fchamber-of-horrors-1966.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=78&amp;q=80&amp;f=avif   78w,
                            /_image?/content%2Fassets%2Fstills%2Fchamber-of-horrors-1966.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=156&amp;q=80&amp;f=avif 156w,
                            /_image?/content%2Fassets%2Fstills%2Fchamber-of-horrors-1966.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=312&amp;q=80&amp;f=avif 312w,
                            /_image?/content%2Fassets%2Fstills%2Fchamber-of-horrors-1966.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=624&amp;q=80&amp;f=avif 624w
                          "
                          src="/_image?/content%2Fassets%2Fstills%2Fchamber-of-horrors-1966.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=312&amp;h=176&amp;q=80&amp;f=avif"
                          alt="A still from Chamber of Horrors (1966)"
                          loading="lazy"
                          decoding="async"
                          width="312"
                          height="175.5"
                          sizes="(min-width: 706px) 312px, (min-width: 1280) 25vw, (min-width: 1472px) 312px, 50vw"
                          class="aspect-[1.77777778] h-auto"
                      /></a>
                    </div>
                    <div class="row-start-2 mb-2 tablet:m-0 tablet:px-6">
                      <a
                        href="/reviews/chamber-of-horrors-1966/"
                        class="block text-md text-default"
                        >Chamber of Horrors<!-- -->
                        <span class="text-sm font-light leading-none text-muted"
                          >1966</span
                        ></a
                      >
                    </div>
                    <div class="mb-6 tablet:m-0 tablet:px-6 tablet:pb-6">
                      <picture
                        ><source
                          srcset="/svg/2-stars-dark.svg"
                          media="(prefers-color-scheme: dark)" />
                        <img
                          src="/svg/2-stars.svg"
                          alt="D+: 2 stars (out of 5)"
                          height="16"
                          width="80"
                      /></picture>
                    </div>
                    <div class="tablet:px-6">
                      <div class="text-sm leading-4 tracking-0.5px text-subtle">
                        <span>Crime</span
                        ><span>
                          |
                          <!-- -->Horror</span
                        ><span>
                          |
                          <!-- -->Thriller</span
                        >
                      </div>
                    </div>
                  </li>
                  <li
                    class="flow-root w-full px-gutter py-6 even:bg-subtle tablet:grid tablet:max-w-[312px] tablet:grid-rows-[auto_auto_1fr_auto] tablet:gap-y-2 tablet:overflow-hidden tablet:rounded-lg tablet:bg-default tablet:p-0 tablet:pb-8 tablet:shadow-all tablet:shadow-border tablet:even:bg-default"
                  >
                    <div class="row-start-1 row-end-1">
                      <a
                        href="/reviews/two-for-the-seesaw-1962/"
                        class="safari-border-radius-fix float-right mb-2 ml-6 block w-[calc(50%_-_12px)] max-w-[312px] overflow-hidden rounded-lg tablet:float-none tablet:m-0 tablet:w-auto tablet:rounded-none"
                        ><img
                          srcset="
                            /_image?/content%2Fassets%2Fstills%2Ftwo-for-the-seesaw-1962.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=78&amp;q=80&amp;f=avif   78w,
                            /_image?/content%2Fassets%2Fstills%2Ftwo-for-the-seesaw-1962.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=156&amp;q=80&amp;f=avif 156w,
                            /_image?/content%2Fassets%2Fstills%2Ftwo-for-the-seesaw-1962.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=312&amp;q=80&amp;f=avif 312w,
                            /_image?/content%2Fassets%2Fstills%2Ftwo-for-the-seesaw-1962.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=624&amp;q=80&amp;f=avif 624w
                          "
                          src="/_image?/content%2Fassets%2Fstills%2Ftwo-for-the-seesaw-1962.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=312&amp;h=176&amp;q=80&amp;f=avif"
                          alt="A still from Two for the Seesaw (1962)"
                          loading="lazy"
                          decoding="async"
                          width="312"
                          height="175.5"
                          sizes="(min-width: 706px) 312px, (min-width: 1280) 25vw, (min-width: 1472px) 312px, 50vw"
                          class="aspect-[1.77777778] h-auto"
                      /></a>
                    </div>
                    <div class="row-start-2 mb-2 tablet:m-0 tablet:px-6">
                      <a
                        href="/reviews/two-for-the-seesaw-1962/"
                        class="block text-md text-default"
                        >Two for the Seesaw<!-- -->
                        <span class="text-sm font-light leading-none text-muted"
                          >1962</span
                        ></a
                      >
                    </div>
                    <div class="mb-6 tablet:m-0 tablet:px-6 tablet:pb-6">
                      <picture
                        ><source
                          srcset="/svg/3-stars-dark.svg"
                          media="(prefers-color-scheme: dark)" />
                        <img
                          src="/svg/3-stars.svg"
                          alt="C+: 3 stars (out of 5)"
                          height="16"
                          width="80"
                      /></picture>
                    </div>
                    <div class="tablet:px-6">
                      <div class="text-sm leading-4 tracking-0.5px text-subtle">
                        <span>Drama</span
                        ><span>
                          |
                          <!-- -->Romance</span
                        >
                      </div>
                    </div>
                  </li>
                  <li
                    class="flow-root w-full px-gutter py-6 even:bg-subtle tablet:grid tablet:max-w-[312px] tablet:grid-rows-[auto_auto_1fr_auto] tablet:gap-y-2 tablet:overflow-hidden tablet:rounded-lg tablet:bg-default tablet:p-0 tablet:pb-8 tablet:shadow-all tablet:shadow-border tablet:even:bg-default"
                  >
                    <div class="row-start-1 row-end-1">
                      <a
                        href="/reviews/the-man-with-nine-lives-1940/"
                        class="safari-border-radius-fix float-right mb-2 ml-6 block w-[calc(50%_-_12px)] max-w-[312px] overflow-hidden rounded-lg tablet:float-none tablet:m-0 tablet:w-auto tablet:rounded-none"
                        ><img
                          srcset="
                            /_image?/content%2Fassets%2Fstills%2Fthe-man-with-nine-lives-1940.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=78&amp;q=80&amp;f=avif   78w,
                            /_image?/content%2Fassets%2Fstills%2Fthe-man-with-nine-lives-1940.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=156&amp;q=80&amp;f=avif 156w,
                            /_image?/content%2Fassets%2Fstills%2Fthe-man-with-nine-lives-1940.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=312&amp;q=80&amp;f=avif 312w,
                            /_image?/content%2Fassets%2Fstills%2Fthe-man-with-nine-lives-1940.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=624&amp;q=80&amp;f=avif 624w
                          "
                          src="/_image?/content%2Fassets%2Fstills%2Fthe-man-with-nine-lives-1940.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=312&amp;h=176&amp;q=80&amp;f=avif"
                          alt="A still from The Man with Nine Lives (1940)"
                          loading="lazy"
                          decoding="async"
                          width="312"
                          height="175.5"
                          sizes="(min-width: 706px) 312px, (min-width: 1280) 25vw, (min-width: 1472px) 312px, 50vw"
                          class="aspect-[1.77777778] h-auto"
                      /></a>
                    </div>
                    <div class="row-start-2 mb-2 tablet:m-0 tablet:px-6">
                      <a
                        href="/reviews/the-man-with-nine-lives-1940/"
                        class="block text-md text-default"
                        >The Man with Nine Lives<!-- -->
                        <span class="text-sm font-light leading-none text-muted"
                          >1940</span
                        ></a
                      >
                    </div>
                    <div class="mb-6 tablet:m-0 tablet:px-6 tablet:pb-6">
                      <picture
                        ><source
                          srcset="/svg/2-half-stars-dark.svg"
                          media="(prefers-color-scheme: dark)" />
                        <img
                          src="/svg/2-half-stars.svg"
                          alt="C-: 2.5 stars (out of 5)"
                          height="16"
                          width="80"
                      /></picture>
                    </div>
                    <div class="tablet:px-6">
                      <div class="text-sm leading-4 tracking-0.5px text-subtle">
                        <span>Horror</span
                        ><span>
                          |
                          <!-- -->Mystery</span
                        ><span>
                          |
                          <!-- -->Sci-Fi</span
                        >
                      </div>
                    </div>
                  </li>
                  <li
                    class="flow-root w-full px-gutter py-6 even:bg-subtle tablet:grid tablet:max-w-[312px] tablet:grid-rows-[auto_auto_1fr_auto] tablet:gap-y-2 tablet:overflow-hidden tablet:rounded-lg tablet:bg-default tablet:p-0 tablet:pb-8 tablet:shadow-all tablet:shadow-border tablet:even:bg-default"
                  >
                    <div class="row-start-1 row-end-1">
                      <a
                        href="/reviews/burial-ground-the-nights-of-terror-1981/"
                        class="safari-border-radius-fix float-right mb-2 ml-6 block w-[calc(50%_-_12px)] max-w-[312px] overflow-hidden rounded-lg tablet:float-none tablet:m-0 tablet:w-auto tablet:rounded-none"
                        ><img
                          srcset="
                            /_image?/content%2Fassets%2Fstills%2Fburial-ground-the-nights-of-terror-1981.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=78&amp;q=80&amp;f=avif   78w,
                            /_image?/content%2Fassets%2Fstills%2Fburial-ground-the-nights-of-terror-1981.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=156&amp;q=80&amp;f=avif 156w,
                            /_image?/content%2Fassets%2Fstills%2Fburial-ground-the-nights-of-terror-1981.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=312&amp;q=80&amp;f=avif 312w,
                            /_image?/content%2Fassets%2Fstills%2Fburial-ground-the-nights-of-terror-1981.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=624&amp;q=80&amp;f=avif 624w
                          "
                          src="/_image?/content%2Fassets%2Fstills%2Fburial-ground-the-nights-of-terror-1981.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=312&amp;h=176&amp;q=80&amp;f=avif"
                          alt="A still from Burial Ground: The Nights of Terror (1981)"
                          loading="lazy"
                          decoding="async"
                          width="312"
                          height="175.5"
                          sizes="(min-width: 706px) 312px, (min-width: 1280) 25vw, (min-width: 1472px) 312px, 50vw"
                          class="aspect-[1.77777778] h-auto"
                      /></a>
                    </div>
                    <div class="row-start-2 mb-2 tablet:m-0 tablet:px-6">
                      <a
                        href="/reviews/burial-ground-the-nights-of-terror-1981/"
                        class="block text-md text-default"
                        >Burial Ground: The Nights of Terror<!-- -->
                        <span class="text-sm font-light leading-none text-muted"
                          >1981</span
                        ></a
                      >
                    </div>
                    <div class="mb-6 tablet:m-0 tablet:px-6 tablet:pb-6">
                      <picture
                        ><source
                          srcset="/svg/3-stars-dark.svg"
                          media="(prefers-color-scheme: dark)" />
                        <img
                          src="/svg/3-stars.svg"
                          alt="C: 3 stars (out of 5)"
                          height="16"
                          width="80"
                      /></picture>
                    </div>
                    <div class="tablet:px-6">
                      <div class="text-sm leading-4 tracking-0.5px text-subtle">
                        <span>Horror</span>
                      </div>
                    </div>
                  </li>
                  <li
                    class="col-[1_/_-1] block px-gutter py-4 text-right shadow-bottom shadow-border tablet:absolute tablet:right-0 tablet:top-0 tablet:shadow-none desktop:right-[var(--gutter-width)]"
                  >
                    <a class="text-accent" href="/reviews/"
                      >All<!-- -->
                      <span class="inline tablet:hidden desktop:inline"
                        >Reviews</span
                      >
                      <!-- -->â†’</a
                    >
                  </li>
                </ul>
              </nav>
            </div>
          </main>
        </div>
        <footer
          class="flex flex-col items-center gap-y-6 bg-[url(&#x27;/assets/ripnotcomingsoon.avif&#x27;)] px-pageMargin py-8 text-inverse"
        >
          <ul
            class="flex w-full flex-wrap justify-center gap-x-4 gap-y-2 text-inverse tablet:gap-x-6 max:w-auto"
          >
            <li class="block whitespace-nowrap tracking-0.5px">
              <a class="text-inverse" href="/">Home</a>
            </li>
            <li class="block whitespace-nowrap tracking-0.5px">
              <a class="text-inherit" href="/how-i-grade/">How I Grade</a>
            </li>
            <li class="block whitespace-nowrap tracking-0.5px">
              <a class="text-inherit" href="/reviews/">Reviews</a>
            </li>
            <li class="block whitespace-nowrap tracking-0.5px">
              <a class="text-inherit" href="/viewings/">Viewing Log</a>
            </li>
            <li class="block whitespace-nowrap tracking-0.5px">
              <a class="text-inherit" href="/cast-and-crew/">Cast &amp; Crew</a>
            </li>
            <li class="block whitespace-nowrap tracking-0.5px">
              <a class="text-inherit" href="/collections/">Collections</a>
            </li>
            <li class="block whitespace-nowrap tracking-0.5px">
              <a class="text-inherit" href="/watchlist/">Watchlist</a>
            </li>
          </ul>
          <p class="text-sm font-light leading-4">
            All stills used in accordance with the<!-- -->
            <a
              href="http://www.copyright.gov/title17/92chap1.html#107"
              class="text-inherit"
              >Fair Use Law.</a
            >
          </p>
          <a href="#top" class="sr-only">To the top â†‘</a>
        </footer>
      </div>
    </div>
    <dialog
      class="relative m-0 h-screen max-h-none w-screen max-w-unset border border-default bg-canvas shadow-lg open:flex open:flex-col tablet:mx-auto tablet:mt-16 tablet:h-min tablet:max-h-[min(100vh_-_128px,900px)] tablet:min-h-60 tablet:max-w-[min(100vw_-_2_*_var(--gutter-width),36rem)] tablet:rounded-xl desktop:min-h-0"
      aria-label="Search"
    >
      <div data-dialog-frame class="contents">
        <button data-close-modal>Cancel</button>
        <div class="contents"><div id="pagefind__search"></div></div>
        <div class="hidden px-gutter py-4 desktop:block">
          <kbd class="rounded-md bg-default p-2 font-mono text-xs text-subtle"
            >Tab</kbd
          >
          to navigate
          <kbd class="rounded-md bg-subtle p-2 font-mono text-xs text-subtle"
            >ESC</kbd
          >
          to close
        </div>
      </div>
    </dialog>
    <script>
      (() => {
        const openBtn = document.querySelector("button[data-open-modal]");
        const shortcut = openBtn?.querySelector("kbd");
        if (!openBtn || !(shortcut instanceof HTMLElement)) return;
        const platformKey = shortcut.querySelector("kbd");
        if (platformKey && /(Mac|iPhone|iPod|iPad)/i.test(navigator.platform)) {
          platformKey.textContent = "âŒ˜";
          openBtn.setAttribute("aria-keyshortcuts", "Meta+K");
        }
        shortcut.style.display = "";
      })();
    </script>
  </body>
</html>
